src/state/reducers/students-state.ts

```diff
@@ -0,0 +1,59 @@
+import { nanoid } from 'nanoid'
+
+import {
+  STUDENTS_ADD,
+  STUDENTS_EDIT,
+  STUDENTS_REMOVE,
+  RESET_STUDENTS
+} from "../actions/students";
+
+interface StudentsState {
+  students: Array<{
+    name: string,
+    id: number,
+    count: number
+  }>
+}
+
+const initialState: StudentsState = { students: [] };
+
+export default (state = initialState, action: StudentsTypes): StudentsState => {
+  switch (action.type) {
+    case STUDENTS_ADD:
+      return {
+        ...state,
+        students: [...state.students, {name: action.name, id: nanoid(), count: 0}]
```

Comment:
`nanoid` creates strings and the `students` array has `id` defined as `number` - let's change the array to use `string` type instead
```ts
interface StudentsState {
  students: Array<{
    name: string,
    id: string,
    count: number
  }>
}
```